<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Lab3</class>
 <widget class="QWidget" name="Lab3">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>430</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QGroupBox" name="fermat">
     <property name="title">
      <string>Факторизація Ферма</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="2">
       <widget class="QLCDNumber" name="fermatFact2"/>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="fermatPushButton">
        <property name="text">
         <string>Розрахувати</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <widget class="QSpinBox" name="fermatInput">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>999999999</number>
        </property>
        <property name="value">
         <number>774311639</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLCDNumber" name="fermatFact1"/>
      </item>
      <item row="1" column="3">
       <widget class="QLabel" name="labelEq">
        <property name="text">
         <string>=</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="labelMult">
        <property name="text">
         <string>x</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QLCDNumber" name="fermatMult"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="perceptron">
     <property name="title">
      <string>Перцептрон</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="2" column="0">
       <widget class="QLabel" name="labelAccuracy">
        <property name="text">
         <string>Точність:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="labelLRate">
        <property name="text">
         <string>Швидкість навчання:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QSpinBox" name="pThreshold">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QSpinBox" name="pDeadline">
        <property name="minimum">
         <number>100</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="labelDeadline">
        <property name="text">
         <string>Дедлайн</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="4">
       <widget class="QLCDNumber" name="pAccuracy"/>
      </item>
      <item row="1" column="0">
       <widget class="QDoubleSpinBox" name="pLearningRate">
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="minimum">
         <double>0.001000000000000</double>
        </property>
        <property name="maximum">
         <double>0.300000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.001000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="labelThreshold">
        <property name="text">
         <string>Поріг</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QPushButton" name="pPushButton">
        <property name="text">
         <string>Розрахувати</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="diaphantus">
     <property name="title">
      <string>Діафантове рівняння</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="1">
       <widget class="QLabel" name="aLabel">
        <property name="text">
         <string>a +</string>
        </property>
       </widget>
      </item>
      <item row="0" column="7">
       <widget class="QLabel" name="dLabel">
        <property name="text">
         <string>d =</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4" colspan="2">
       <widget class="QLCDNumber" name="cLcdNumber"/>
      </item>
      <item row="3" column="2" colspan="2">
       <widget class="QLabel" name="bResLabel">
        <property name="text">
         <string>b</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignHCenter|Qt::AlignTop</set>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QLabel" name="cLabel">
        <property name="text">
         <string>c +</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLabel" name="bLabel">
        <property name="text">
         <string>b +</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QLabel" name="aResLabel">
        <property name="text">
         <string>a</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignHCenter|Qt::AlignTop</set>
        </property>
       </widget>
      </item>
      <item row="2" column="2" colspan="2">
       <widget class="QLCDNumber" name="bLcdNumber"/>
      </item>
      <item row="3" column="4" colspan="2">
       <widget class="QLabel" name="cResLabel">
        <property name="text">
         <string>c</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignHCenter|Qt::AlignTop</set>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QLCDNumber" name="aLcdNumber"/>
      </item>
      <item row="2" column="6" colspan="2">
       <widget class="QLCDNumber" name="dLcdNumber"/>
      </item>
      <item row="3" column="6" colspan="2">
       <widget class="QLabel" name="dResLabel">
        <property name="text">
         <string>d</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignHCenter|Qt::AlignTop</set>
        </property>
       </widget>
      </item>
      <item row="1" column="4" colspan="4">
       <widget class="QPushButton" name="dPushButton">
        <property name="text">
         <string>Розрахувати</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="4">
       <widget class="QDoubleSpinBox" name="eqSpinBox">
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="value">
         <double>20.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QSpinBox" name="aSpinBox">
        <property name="value">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QSpinBox" name="bSpinBox">
        <property name="value">
         <number>2</number>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QSpinBox" name="cSpinBox">
        <property name="value">
         <number>3</number>
        </property>
       </widget>
      </item>
      <item row="0" column="6">
       <widget class="QSpinBox" name="dSpinBox">
        <property name="value">
         <number>4</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
